README.md

```diff
@@ -1,53 +1,251 @@
-
-* `git clone <repository-url>` this repository
-* change into the new directory
-* `npm install`
-* `bower install`
-
-## Running / Development
-
-* `ember server`
-* Visit your app at [http://localhost:4200](http://localhost:4200).
-
-### Code Generators
-
-Make use of the many generators for code, try `ember help generate` for more details
-
-### Running Tests
-
-* `ember test`
-* `ember test --server`
-
-### Building
-
-* `ember build` (development)
-* `ember build --environment production` (production)
-
-### Deploying
-
-Specify what it takes to deploy your app.
-
-## Further Reading / Useful Links
-
-* [ember.js](http://emberjs.com/)
-* [ember-cli](http://www.ember-cli.com/)
-* Development Browser Extensions
-  * [ember inspector for chrome](https://chrome.google.com/webstore/detail/ember-inspector/bmdblncegkenkacieihfhpjfppoconhi)
-  * [ember inspector for firefox](https://addons.mozilla.org/en-US/firefox/addon/ember-inspector/)
-
+# Impagination
+
+[![npm version](https://badge.fury.io/js/impagination.svg)](https://badge.fury.io/js/impagination)
+[![frontside approved](https://zm6o5z0kfa.execute-api.us-east-1.amazonaws.com/dev/badges/frontside/)](https://frontside.io)
+
+Put the *fun* back in asynchronous, paged, datasets.
+
+Whatever your use-case: infinite scrolling lists, a carousel browser,
+or even a classic page-by-page result list, Impagination frees you to
+focus on what you want to do with your data, not the micro-logistics
+of when to fetch it. All you provide Impagination is the logic to
+fetch a single page, plus how many pages you want it to pre-fetch
+ahead of you, and it will figure out the rest.
+
+Impagination is built using an event-driven immutable style, so it is
+ideal for use with UI frameworks like Ember, Angular, or React. That
+said, it has zero dependencies apart from JavaScript, so it can be
+used from node as well.
+
+## Usage
+
+To get started, create a dataset. There are only two required parameters
+`fetch`, and `observe`:
+
+```javascript
+import { Dataset } from 'impagination';
+
+let state = null;
+
+let dataset = new Dataset({
+  // how many records should we "keep ahead"? (default = pageSize)
+  loadHorizon: 10,
+  // fetch in pages of 5 (default 10)
+  pageSize: 5,
+  // this fake fetch function returns a page where the "records" *are*
+  // the offsets
+  fetch: function(pageOffset, pageSize, stats) {
+    stats.totalPages = 5;
+    return new Promise(function(resolve) {
+      resolve(return new Array(pageSize).fill(0).map(function(zero, i)) {
+        return pageOffset * pageSize + i;
+      });
+    });
+  },
+  //this function is invoked whenever a new state is generated.
+  observe: function(nextState) {
+    state = nextState;
+  }
+});
+```
+
+This will emit a state immediately, however this state will not have
+anything in it, but That's because we haven't told the dataset where we
+want to start reading from.
+
+```javascript
+state.length //=> 0;
+state.get(0) //=> null;
+```
```

Comment:
We should add issues to the repo for things that need to be done so it's easier for those to come in and pick something up